cli/azd/extensions/microsoft.ai.agents/ci-build.ps1

```diff
@@ -0,0 +1,144 @@
+    # will begin either a module path@version (when using modules),
+    # or a plain import path (when using the standard library, or GOPATH).
+    "-trimpath",
+
+    # Use buildmode=pie (Position Independent Executable) for enhanced security across platforms
+    # against memory corruption exploits across all major platforms.
+    #
+    # On Windows, the -buildmode=pie flag enables Address Space Layout 
+    # Randomization (ASLR) and automatically sets DYNAMICBASE and HIGH-ENTROPY-VA flags in the PE header.
+    "-buildmode=pie"
+)
+
+if ($CodeCoverageEnabled) {
+    $buildFlags += "-cover"
+}
+
+# Build constraint tags
+# cfi: Enable Control Flow Integrity (CFI),
+# cfg: Enable Control Flow Guard (CFG),
+# osusergo: Optimize for OS user accounts
+$tagsFlag = "-tags=cfi,cfg,osusergo"
+
+# ld linker flags
+# -s: Omit symbol table and debug information
+# -w: Omit DWARF symbol table
+# -X: Set variable at link time. Used to set the version in source.
+
+# TODO: set version properly
+$ldFlag = "-ldflags=-s -w -X 'github.com/azure/azure-dev/cli/azd/internal.Version=$Version (commit $SourceVersion)' "
+
+if ($IsWindows) {
+    $msg = "Building for Windows"
+    Write-Host $msg
+}
+elseif ($IsLinux) {
+    Write-Host "Building for linux"
+}
+elseif ($IsMacOS) {
+    Write-Host "Building for macOS"
+}
+
+# Add output file flag based on specified output file name
+$outputFlag = "-o=$OutputFileName"
+
+# collect flags
+$buildFlags += @(
+    $tagsFlag,
+    $ldFlag,
+    $outputFlag
+)
+
+function PrintFlags() {
+    param(
+        [string] $flags
+    )
+
+    # Attempt to format flags so that they are easily copy-pastable to be ran inside pwsh
+    $i = 0
+    foreach ($buildFlag in $buildFlags) {
+        # If the flag has a value, wrap it in quotes. This is not required when invoking directly below,
+        # but when repasted into a shell for execution, the quotes can help escape special characters such as ','.
+        $argWithValue = $buildFlag.Split('=', 2)
+        if ($argWithValue.Length -eq 2 -and !$argWithValue[1].StartsWith("`"")) {
+            $buildFlag = "$($argWithValue[0])=`"$($argWithValue[1])`""
+        }
+
+        # Write each flag on a newline with '`' acting as the multiline separator
+        if ($i -eq $buildFlags.Length - 1) {
+            Write-Host "  $buildFlag"
+        }
+        else {
+            Write-Host "  $buildFlag ``"
+        }
+        $i++
+    }
+}
+
+$oldGOEXPERIMENT = $env:GOEXPERIMENT
+# Enable the loopvar experiment, which makes the loop variaible for go loops like `range` behave as most folks would expect.
+# the go team is exploring making this default in the future, and we'd like to opt into the behavior now.
+$env:GOEXPERIMENT = "loopvar"
+
+try {
+    Write-Host "Running: go build ``"
+    PrintFlags -flags $buildFlags
+    go build @buildFlags
+    if ($LASTEXITCODE) {
+        Write-Host "Error running go build"
+        exit $LASTEXITCODE
+    }
+
+    if ($BuildRecordMode) {
```

Comment:
What gets enabled when you use this? There's no documentation on the script parameter and since it's a tag I assume it's activating something in the extension code, or maybe one of our dependencies (like azd itself?)

Reply:
I know azd uses this in azd's [ci-build.ps1](https://github.com/Azure/azure-dev/blob/main/cli/azd/ci-build.ps1) to enable specific recording functionality, but I'm unsure yet what this means for extensions.

Reply:
Tracking this in https://github.com/Azure/azure-dev/issues/5751